Assets/Mirror/Examples/AdditiveScenes/Scripts/ShootingTankBehaviour.cs

```diff
@@ -0,0 +1,65 @@
+using UnityEngine;
+
+namespace Mirror.Examples.Additive
+{
+    // This script demonstrates the NetworkAnimator and how to leverage
+    // the built-in observers system to track players.
+    // Note that all ProximityCheckers should be restricted to the Player layer.
+    public class ShootingTankBehaviour : NetworkBehaviour
+    {
+        [SyncVar]
+        public Quaternion Rotation;
+
+        NetworkAnimator networkAnimator;
+
+        [ServerCallback]
+        void Start()
+        {
+            networkAnimator = GetComponent<NetworkAnimator>();
+        }
+
+        [Range(0, 1)]
+        public float turnSpeed = 0.1f;
+
+        void Update()
+        {
+            if (isServer && netIdentity.observers.Count > 0)
+                ShootNearestPlayer();
+
+            if (isClient)
+                transform.rotation = Quaternion.Slerp(transform.rotation, Rotation, turnSpeed);
+        }
+
+        NetworkConnection[] networkConnections = new NetworkConnection[] { };
+
+        [Server]
+        void ShootNearestPlayer()
+        {
+            GameObject target = null;
+            float distance = 100f;
+
+            // Avoid a foreach by copying the observers to an array for iteration
+            System.Array.Resize(ref networkConnections, netIdentity.observers.Count);
+            netIdentity.observers.Values.CopyTo(networkConnections, 0);
+
+            for (int i = 0; i < networkConnections.Length; i++)
```

Comment:
foreach should be easier here

Reply:
Yes it would but this is in a hot path.

Reply:
I'm pretty sure a foreach would be faster than resizing and copying all elements into an array and then iterating. Either way the original structure is being iterated.

Reply:
If you all are fine with foreach than I'm good with it.

Reply:
foreach as requested